lib/netutil/conn.go

```diff
@@ -27,19 +27,19 @@ type connMetrics struct {
 }
 
 func (cm *connMetrics) init(group, name, addr string) {
-	cm.readCalls = metrics.NewCounter(fmt.Sprintf(`%s_read_calls_total{name=%q, addr=%q}`, group, name, addr))
-	cm.readBytes = metrics.NewCounter(fmt.Sprintf(`%s_read_bytes_total{name=%q, addr=%q}`, group, name, addr))
-	cm.readErrors = metrics.NewCounter(fmt.Sprintf(`%s_errors_total{name=%q, addr=%q, type="read"}`, group, name, addr))
-	cm.readTimeouts = metrics.NewCounter(fmt.Sprintf(`%s_read_timeouts_total{name=%q, addr=%q}`, group, name, addr))
+	cm.readCalls = metrics.GetOrCreateCounter(fmt.Sprintf(`%s_read_calls_total{name=%q, addr=%q}`, group, name, addr))
```

Comment:
Could you comment why this change is needed?

Reply:
emmm, this is not so needed here, I will make another pr if necessary

In my option, GetOrCreateCounter is more safe than NewCounter, as GetOrCreateCounter can run multi times and NewCounter will panic if do more than one time

Reply:
`metrics.NewCounter` is used intentionally in places where it is expected that they are executed only once. This allows immediately catching the bug via `panic` when the code is executed multiple times.